src/validation.cpp

```diff
@@ -5046,11 +5046,14 @@ bool LoadMempool(CTxMemPool& pool)
             file >> nFeeDelta;
 
             CAmount amountdelta = nFeeDelta;
+            if (!MoneyRange(amountdelta)) {
+                return false;
+            }
```

Comment:
There is no requirement for deltas to be in `MoneyRange`... Better use a numeric_limit max

Reply:
Before I push the commit does this look ok?

```
CAmount amountdelta = nFeeDelta;
            if (amountdelta > numeric_limits<int64_t>::max() || amountdelta < numeric_limits<int64_t>::min())  {
                return false;
            }
```

Reply:
@luke-jr Oh, crap thanks for alerting me about that! I naïvely assumed that the bounds of `CAmount` were defined by `MoneyRange(…)`, but I now understand that this is a case where negative amounts are allowed for a `CAmount`. I assume the valid range here is `[-MAX_MONEY, MAX_MONEY]` then?

Should `CAmount` perhaps be reserved for cases where `MoneyRange(…)` defines the bounds?

@rajarshimaitra I don't think that is what @luke-jr is suggesting: note that `CAmount` is `int64_t` so `amountdelta` is guaranteed to be in the range you're checking for in the suggested diff (`[std:: numeric_limits<int64_t>::min(), std:: numeric_limits<int64_t>::max()]`) :)

Reply:
Oh, right. Well, this sounds quite deep for my grasp and I wouldn't know what exactly to do. Do suggest and will update the PR accordingly.